lib/nimble_options.ex

```diff
@@ -131,6 +131,13 @@ defmodule NimbleOptions do
       an updated value, then that updated value is used in the resulting list. Validation
       fails at the *first* element that is invalid according to `subtype`.
 
+    * `{:tuple, tuple_of_subtypes}` - A tuple as described by `tuple_of_subtypes`.
```

Comment:
I wonder if it should be `tuple_of_subtypes` or `list_of_subtypes`. I think list would be better because it can be any size really. But other than that, :shipit: !

Reply:
@josevalim well the second element of `{:tuple, ...}` should be a *tuple* itself. The idea there was that we can "match" the tuple of subtypes with the actual value to validate. For example, you'd say `{:tuple, {:int, :int, :string}}` to validate tuples like `{1, 2, "three"}`. Are you referring to just the name of `tuple_of_subtypes` in the docs, or you want to make that an actual list?

Reply:
An actual list to represent the tuple elements. The fact you need to call Tuple.to_list kinda tells you a tuple is not the correct type. There is also no way in typespecs to say you expect a tuple of any size made of a given type, etc.

I see where you are coming from but the best way to represent the second argument is a list because it can be of any size.